### DISCUSSION TITLE
java有没有什么好的办法可以高效率解析messagechain？

### DISCUSSION THREAD

**Discussion by Zqysl** [mamoe/mirai#1276] [Q&A]
比如我的bot支持如下群管命令：禁言[@某人][分钟]

有管理员在群中发送：”禁言@某群员60“

有没有可以直接通过messagechain分辨这是一条群管命令，并提取出"@某群员"和"60"的方法呢

**Comment by Him188** [comment]
没有内置的方法

**Reply by Zqysl** [reply]
能否提供下实现思路，不经过mirai码的情况下

**Reply by Karlatemp** [reply]
MessageChain 实现 `java.util.List`
按照正常检测 List 的方法就行

**Comment by eiriksgata** [comment]
自己用正则表达式匹配一下就好了
